provider/virtual_network/network_base.py

```
@@ -39,16 +41,16 @@ def _get_vm_ip():
                            and addr.get('mngtmpaddr') is not True]
 
         if len(target_addr) == 0:
-            LOG.warn(f'No ip addr of given mac: {mac}')
+            LOG.warning(f'No ip addr of given mac: {mac}')
             return
         elif len(target_addr) > 1:
-            LOG.warn(f'Multiple ip addr: {target_addr}')
+            LOG.warning(f'Multiple ip addr: {target_addr}')
 
         return target_addr[0]['local']
 
     vm_ip = utils_misc.wait_for(_get_vm_ip, timeout, ignore_errors=True)
```

May I ask if you need to replace 'ignore_errors=True' with 'ignore_errors=ignore_error'?

---

After checking the history and refreshing my memory, I found that the 'wait_for' was supposed to extend waiting time when trying to get ip address. Therefore ignoring errors when not getting ip during that time. 
The newly added arg 'ignore_error' is supposed to control the behavior of the function when not getting ip address after timeout.